src/Commands/RsyncCommand.php

```diff
@@ -92,8 +92,28 @@ protected function rsync($site_env_id, $src, $dest, array $rsyncOptions)
             $rsyncOptionString = "$rsyncOptionString --temp-dir=$tmpdir --delay-updates";
         }
 
+        // Add in a partial option if one was not already specified
+        if (!preg_match('/(^| )--partial/', $rsyncOptionString)) {
+            $rsyncOptionString = "$rsyncOptionString --partial";
+        }
+
         $this->log()->notice('Running {cmd}', ['cmd' => "rsync $rsyncOptionString $src $dest"]);
-        $this->passthru("rsync $rsyncOptionString --ipv4 --exclude=.git -e 'ssh -p 2222' '$src' '$dest' ");
+        $attempts = 1;
+        $maxAttempts = 5;
+        while (true) {
```

Comment:
tip: in order to avoid `while (true)` and an extra `if`:

```php
$attempts = 0;
$maxAttempts = 5;
while (++$attempts <= $maxAttempts) {
    try {
        $this->passthru("rsync $rsyncOptionString --ipv4 --exclude=.git -e 'ssh -p 2222' '$src' '$dest' ");
        $this->log()->notice('Command finished successfully.');
        return;
    } catch (TerminusException $e) {
        $this->log()->notice('Retrying rsync command...');
    }
}

$this->log()->notice('Command failed after {attempts} attempts.', ['attempts' => $maxAttempts]);
throw $e;
```

Reply:
lol yeah. I started this with unlimited attempts, then quickly changed to a max number of attempts and didn't update the rest of the code. Thanks!